esrally/track/params.py

```diff
@@ -936,9 +936,11 @@ def create_readers(
     create_reader,
 ):
     logger = logging.getLogger(__name__)
-    readers = []
-    for corpus in corpora:
-        for docs in corpus.documents:
+    # pre-initialize in order to assign parallel tasks for different corpora through assignment
+    readers = [None for corpus in corpora for _ in corpus.documents] * num_clients
+    # stagger which corpus each client starts with for better parallelism
+    for group, corpus in enumerate(corpora[(start_client_index + mod) % len(corpora)] for mod in range(len(corpora))):
```

Comment:
primarily looking for feedback on the round-robin approach implemented between this line and line 961 below. I feel like it's unreadable but was stumped with regards making it better

Reply:
I think this approach makes sense. As you correctly identified, there are tracks like `http_logs` that currently have the assumption that specific corpus files contain a certain timerange of events and to run them in a realistic way we'd need the existing sequential strategy or change the corpora files in the track.

I'd be interested to see the impact of this approach also in the solutions/logs tracks.